# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/angle/checkout/src/libANGLE/renderer/d3d/FramebufferD3D.cpp
# Commit: 3a91885d47e0
# Full Hash: 3a91885d47e0ef7fba9749562d52300eea67c651
# Author: Miko Mynttinen <mikokm@gmail.com>
# Date: 2019-11-11 21:52:52
# Regressor Bug: 1578576
# File Overlap Count: 4
# Description:
#   Bug 1578576 - Part 2: Update to ANGLE 3865 r=jgilbert
#   
#   Depends on D44561
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D44579
# ==============================================================================

diff -r 77cee77c0307 -r 3a91885d47e0 gfx/angle/checkout/src/libANGLE/renderer/d3d/FramebufferD3D.cpp
--- a/gfx/angle/checkout/src/libANGLE/renderer/d3d/FramebufferD3D.cpp	Fri Nov 08 16:41:42 2019 +0000
+++ b/gfx/angle/checkout/src/libANGLE/renderer/d3d/FramebufferD3D.cpp	Fri Nov 08 17:06:00 2019 +0000
@@ -69,7 +69,7 @@
     if (mask & GL_DEPTH_BUFFER_BIT)
     {
         if (state.getDepthStencilState().depthMask &&
-            framebufferObject->getDepthbuffer() != nullptr)
+            framebufferObject->getDepthAttachment() != nullptr)
         {
             clearParams.clearDepth = true;
         }
@@ -77,8 +77,8 @@
 
     if (mask & GL_STENCIL_BUFFER_BIT)
     {
-        if (framebufferObject->getStencilbuffer() != nullptr &&
-            framebufferObject->getStencilbuffer()->getStencilSize() > 0)
+        if (framebufferObject->getStencilAttachment() != nullptr &&
+            framebufferObject->getStencilAttachment()->getStencilSize() > 0)
         {
             clearParams.clearStencil = true;
         }
@@ -356,7 +356,7 @@
 
     const auto &colorAttachments = mState.getColorAttachments();
     const auto &drawBufferStates = mState.getDrawBufferStates();
-    const auto &workarounds      = mRenderer->getWorkarounds();
+    const auto &features         = mRenderer->getFeatures();
 
     for (size_t attachmentIndex = 0; attachmentIndex < colorAttachments.size(); ++attachmentIndex)
     {
@@ -370,7 +370,7 @@
                    drawBufferState == (GL_COLOR_ATTACHMENT0_EXT + attachmentIndex));
             colorAttachmentsForRender.push_back(&colorAttachment);
         }
-        else if (!workarounds.mrtPerfWorkaround)
+        else if (!features.mrtPerfWorkaround.enabled)
         {
             colorAttachmentsForRender.push_back(nullptr);
         }
@@ -379,7 +379,7 @@
     // When rendering with no render target on D3D, two bugs lead to incorrect behavior on Intel
     // drivers < 4815. The rendering samples always pass neglecting discard statements in pixel
     // shader. We add a dummy texture as render target in such case.
-    if (mRenderer->getWorkarounds().addDummyTextureNoRenderTarget &&
+    if (mRenderer->getFeatures().addDummyTextureNoRenderTarget.enabled &&
         colorAttachmentsForRender.empty() && activeProgramOutputs.any())
     {
         static_assert(static_cast<size_t>(activeProgramOutputs.size()) <= 32,